includes/Classifai/Providers/Azure/TextToSpeech.php

```diff
@@ -492,6 +495,108 @@ function( $post_type ) {
 		);
 	}
 
+	/**
+	 * Add meta box to post types that support speech synthesis.
+	 *
+	 * @param string $post_type Post type.
+	 */
+	public function add_meta_box( $post_type ) {
+		if ( ! in_array( $post_type, $this->get_supported_post_types(), true ) ) {
+			return;
+		}
+
+		\add_meta_box(
+			'classifai-text-to-speech-meta-box',
+			__( 'ClassifAI Text to Speech Processing', 'classifai' ),
+			[ $this, 'render_meta_box' ],
+			null,
+			'side',
+			'low',
+			array( '__back_compat_meta_box' => true )
+		);
+	}
+
+	/**
+	 * Render meta box content.
+	 *
+	 * @param \WP_Post $post WP_Post object.
+	 */
+	public function render_meta_box( $post ) {
+		wp_nonce_field( 'classifai_text_to_speech_meta_action', 'classifai_text_to_speech_meta' );
+
+		$process_content = get_post_meta( $post->ID, self::SYNTHESIZE_SPEECH_KEY, true );
+		$process_content = ( 'no' === $process_content ) ? 'no' : 'yes';
+
+		$post_type       = get_post_type_object( get_post_type( $post ) );
+		$post_type_label = esc_html__( 'Post', 'classifai' );
+		if ( $post_type ) {
+			$post_type_label = $post_type->labels->singular_name;
+		}
+
+		$audio_id = get_post_meta( $post->ID, self::AUDIO_ID_KEY, true );
+		?>
+
+		<p>
+			<label for="classifai_synthesize_speech">
+				<input type="checkbox" value="yes" id="classifai_synthesize_speech" name="classifai_synthesize_speech" <?php checked( $process_content, 'yes' ); ?> />
+				<?php esc_html_e( 'Enable audio generation', 'classifai' ); ?>
+			</label>
+			<span class="description">
+				<?php
+				/* translators: %s Post type label */
+				printf( esc_html__( 'ClassifAI will generate audio for this %s when it is published or updated', 'classifai' ), esc_html( $post_type_label ) );
+				?>
+			</span>
+		</p>
+
+		<?php
+		if ( $audio_id && wp_get_attachment_url( $audio_id ) ) {
```

Comment:
Maybe only show audio preview if audio generation is enabled?

![image](https://github.com/10up/classifai/assets/10613171/cd264a6b-946b-4c15-ac1b-acb233d4ab6b)

Reply:
Yeah, I debated on this as I'm not sure for a UX perspective it's clear that unchecking this option should also remove the previously generated audio. But that is how we do it right now in the Block Editor so in order to try and keep parity between the two, I've updated this.

Probably worth a follow up at some point to make it more clear in both the Block and Classic Editor that unchecking this box not only won't generate audio anymore but also removes the display of any previously generated audio